client/src/OnDeviceComponent.ts

```diff
@@ -606,8 +610,14 @@ export class OnDeviceComponent {
 			requestBuffers.push(binaryBuffer);
 		}
 
+		if (this.clientSocketPromise) {
+			await this.clientSocketPromise;
+		}
+
 		if (!this.clientSocket) {
-			this.clientSocket = await this.setupClientSocket(options);
+			this.clientSocketPromise = this.setupClientSocket(options);
+			this.clientSocket = await this.clientSocketPromise;
+			this.clientSocketPromise = undefined;
 		}
```

Comment:
Maybe move this into a separate function? Something like `private ensureSocket(options)` ?

Reply:
updated. Let me know what you think